### PR TITLE
Enable raising for assertionless tests for internal framework tests

### PR DESCRIPTION
**Pull request by fatkodima**
Follow up to #51625.

This PR now enables this feature for all the internal tests in the framework.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
After discussing with @dhh I decided to remove the configuration to raise and just make it so we always warns using Ruby.

See https://github.com/rails/rails/commit/6a6c7e64f5bbea9b66c8b93eb85278805c2700a1

**Comment by fatkodima** [comment]
Works for me this way too. So, are you ok with the changes in this PR (if I update them with the latest `main` changes)?

**Comment by rafaelfranca** [comment]
Yes. We should apply the changes here. I'll review it properly

File: actionpack/test/controller/routing_test.rb [hunk 1]
```diff
@@ -513,6 +513,7 @@ def test_named_route_without_hash
         get ":controller/:action/:id", as: "normal"
```
Comment (rafaelfranca) [inline actionpack/test/controller/routing_test.rb:513]: `assert_nothing_raised` here

File: actionpack/test/dispatch/mapper_test.rb [hunk 2]
```diff
@@ -37,6 +37,7 @@ def asts
 
       def test_initialize
         Mapper.new FakeSet.new
```
Comment (rafaelfranca) [inline actionpack/test/dispatch/mapper_test.rb:39]: assert_nothing_raised

File: actionpack/test/dispatch/system_testing/driver_test.rb [hunk 3]
```diff
@@ -32,6 +32,7 @@ class DriverTest < ActiveSupport::TestCase
     original_driver_path = ::Selenium::WebDriver::Chrome::Service.driver_path
     ::Selenium::WebDriver::Chrome::Service.driver_path = "bin/test"
     ActionDispatch::SystemTesting::Driver.new(:selenium, using: :headless_chrome, screen_size: [1400, 1400])
+    pass
```
Comment (rafaelfranca) [inline actionpack/test/dispatch/system_testing/driver_test.rb:35]: assert_nothing_raised

File: actionpack/test/dispatch/system_testing/driver_test.rb [hunk 4]
```diff
@@ -49,6 +50,7 @@ class DriverTest < ActiveSupport::TestCase
     original_driver_path = ::Selenium::WebDriver::Firefox::Service.driver_path
     ::Selenium::WebDriver::Firefox::Service.driver_path = "bin/test"
     ActionDispatch::SystemTesting::Driver.new(:selenium, using: :headless_firefox, screen_size: [1400, 1400])
+    pass
```
Comment (rafaelfranca) [inline actionpack/test/dispatch/system_testing/driver_test.rb:53]: assert_nothing_raised

File: activerecord/test/cases/adapters/postgresql/schema_test.rb [hunk 5]
```diff
@@ -188,6 +188,7 @@ def test_schema_change_with_prepared_stmt
       @connection.exec_query "alter table developers add column zomg int", "sql", []
       altered = true
       @connection.exec_query "select * from developers where id = $1", "sql", [bind_param(1)]
```
Comment (rafaelfranca) [inline activerecord/test/cases/adapters/postgresql/schema_test.rb:190]: assert_nothing_raised

File: activerecord/test/cases/errors_test.rb [hunk 6]
```diff
@@ -12,5 +12,6 @@ def test_can_be_instantiated_with_no_args
     rescue ArgumentError
       raise "Instance of #{error_klass} can't be initialized with no arguments"
     end
+    pass
```
Comment (rafaelfranca) [inline activerecord/test/cases/errors_test.rb:15]: assert_nothing_raised

**Pull request merged**